src/github/pullRequestOverview.ts

```diff
@@ -500,19 +500,47 @@ export class PullRequestOverviewPanel extends IssueOverviewPanel<PullRequestMode
 				});
 			}
 
-			const milestoneToAdd = await vscode.window.showQuickPick(
-				getMilestoneOptions(this._folderRepositoryManager),
-				{
-					canPickMany: false,
-				},
-			);
-
-			if (milestoneToAdd && isMilestoneQuickPickItem(milestoneToAdd)) {
-				await this._item.updateMilestone(milestoneToAdd.id);
-				this._replyMessage(message, {
-					added: milestoneToAdd.milestone,
+			const quickPick = vscode.window.createQuickPick();
+			quickPick.canSelectMany = false;
+			quickPick.placeholder = 'Select a milestone to add';
+			quickPick.buttons = [{
+				iconPath: new vscode.ThemeIcon('add'),
+				tooltip: 'Create',
+			}];
+			quickPick.onDidTriggerButton((_) => {
+				quickPick.hide();
+
+				const inputBox = vscode.window.createInputBox();
+				inputBox.title = 'Create new milestone';
+				inputBox.placeholder = 'New milestone name';
+				if (quickPick.value !== ''){
+					inputBox.value = quickPick.value;
+				}
+				inputBox.show();
+				inputBox.onDidAccept(() => {
+					if (inputBox.value !== ''){
+						this._folderRepositoryManager.createMilestone(this._item.githubRepository, inputBox.value);
```

Comment:
I was thinking about this. On the one hand you select to create a milestone so it might only do that as it's currently the only way to create a milestone, on the other hand, the user clicked on add milestone so it might as well just add it. I'll change it so it adds the milestone after creating it if you agree.

Reply:
Given that the user created the milestone while trying to assign a milestone to a PR, I agree that it makes sense to add the milestone after it's been created.